src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/SelectVariants.java

```diff
@@ -437,7 +438,9 @@
     private Set<String> IDsToKeep = null;
     private Set<String> IDsToRemove = null;
 
-    private final List<Allele> diploidNoCallAlleles = Arrays.asList(Allele.NO_CALL, Allele.NO_CALL);
+    private final List<Allele> diploidNoCallAlleles = GATKVariantContextUtils.noCallAlleles(2);
+
+    private final Map<Integer, Integer> ploidyToNumberOfAlleles = new HashMap<Integer, Integer>();
```

Comment:
dont use HashMap - either treemap or linkedhashmap. We need stable iteration order